src/device/usbd.c

```diff
@@ -1420,4 +1424,25 @@ bool usbd_edpt_iso_activate(uint8_t rhport, tusb_desc_endpoint_t const * desc_ep
   return dcd_edpt_iso_activate(rhport, desc_ep);
 }
 
+static bool usbd_edpt_check_max_count(uint8_t rhport)
+{
+  uint8_t edpt_target_in_max = 0, edpt_target_out_max = 0;
+  dcd_get_edpt_max_count(rhport, &edpt_target_in_max, &edpt_target_out_max);
+  const uint8_t edpt_total_required_count = CFG_TUD_IN_EP_COUNT + CFG_TUD_OUT_EP_COUNT;
+  const uint8_t edpt_target_max = edpt_target_out_max;
+
+  if (edpt_total_required_count > edpt_target_max) {
+    TU_LOG(USBD_DBG, "Number of required endpoints: %d is more than max number of endpoints on target: %d\r\n", edpt_total_required_count, edpt_target_max);
```

Comment:
Let's use log level 1 to signal an error
```suggestion
    TU_LOG(1, "Number of required endpoints: %d is more than max number of endpoints on target: %d\r\n", edpt_total_required_count, edpt_target_max);
```

From tinyusb docs:
```
// CFG_TUSB_DEBUG for debugging
// 0 : no debug
// 1 : print error
// 2 : print warning
// 3 : print info
```

Same on line 1442

Reply:
Is there any motivation to add global macros for `TU_LOG`, and use them instead of numbers? So we would have something like:

`TU_LOG(1, ..)` -> `TU_LOG(ERROR, ..)`
`TU_LOG(2, ..)` -> `TU_LOG(WARNING ..)`

Reply:
We are trying to be as compatible with hathach/tinyusb as possible. So although it is a good idea, I wouldn't introduce any new macros